solidity/test/integration/Slashing.test.ts

```diff
@@ -0,0 +1,531 @@
+            walletMembers.length
+          )
+
+          for (let i = 0; i < walletMembers.length; i++) {
+            const slashing = await staking.slashingQueue(i)
+
+            expect(slashing.amount).to.equal(
+              amountToSlash,
+              `unexpected slashing amount for ${i}`
+            )
+
+            expect(slashing.stakingProvider).to.equal(
+              walletMembers[i].stakingProvider,
+              `unexpected staking provider for ${i}`
+            )
+          }
+        })
+
+        it("should not close the wallet in the wallet registry", async () => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          expect(await walletRegistry.isWalletRegistered(ecdsaWalletID)).to.be
+            .true
+        })
+
+        it("should transition the wallet in the bridge to moving funds state", async () => {
+          const storedWallet = await bridge.wallets(walletPubKeyHash160)
+
+          expect(storedWallet.state).to.be.equal(walletState.MovingFunds)
+        })
+
+        it("should consume around 3 150 000 gas for Bridge.notifyRedemptionTimeout transaction", async () => {
+          await assertGasUsed(notifyRedemptionTimeoutTx, 3_150_000, 100_000)
+        })
+      })
+    })
+  })
+
+  describe("notifyMovingFundsTimeout", async () => {
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      await restoreSnapshot()
+    })
+
+    describe("when wallet is created", async () => {
+      const deposit = SingleP2SHDeposit.deposits[0]
+
+      const walletPubKeyHash160 = deposit.reveal.walletPubKeyHash
+      const { walletPublicKey, walletID: ecdsaWalletID } = deposit.ecdsaWallet
+
+      let walletMembers: Operators
+
+      before(async () => {
+        const requestNewWalletTx = await bridge.requestNewWallet(NO_MAIN_UTXO)
+
+        await produceRelayEntry(walletRegistry, randomBeacon)
+        ;({ walletMembers } = await performEcdsaDkg(
+          walletRegistry,
+          walletPublicKey,
+          requestNewWalletTx.blockNumber
+        ))
+
+        const { fundingTx, reveal } = SingleP2SHDeposit.deposits[0]
+        reveal.vault = tbtcVault.address
+
+        // We use a deposit funding bitcoin transaction with a very low amount,
+        // so we need to update the dust threshold to be below it.
+        await updateDepositDustThreshold(10000) // 0.0001 BTC)
+
+        await bridge.revealDeposit(fundingTx, reveal)
+
+        // TODO: Replace mocks with the real implementation
+        relay.getCurrentEpochDifficulty.returns(
+          SingleP2SHDeposit.chainDifficulty
+        )
+        relay.getPrevEpochDifficulty.returns(SingleP2SHDeposit.chainDifficulty)
+
+        await bridge.submitDepositSweepProof(
+          SingleP2SHDeposit.sweepTx,
+          SingleP2SHDeposit.sweepProof,
+          SingleP2SHDeposit.mainUtxo,
+          tbtcVault.address
+        )
```

Comment:
We do deposit reveal and sweep here in order to have a non-zero BTC balance right? If so, can we drop a comment clarifying that?